docs/source/en/using-diffusers/callback.md

```diff
@@ -19,13 +19,50 @@ The denoising loop of a pipeline can be modified with custom defined functions u
 
 This guide will demonstrate how callbacks work by a few features you can implement with them.
 
+## Official callbacks
+
+We provide a list of callbacks you can plug into an existing pipeline and modify the denoising loop. This is the current list of official callbacks:
+
+- `SDCFGCutoffCallback`: Disables the CFG after a certain number of steps for SD 1.5 pipelines.
+- `SDXLCFGCutoffCallback`: Disables the CFG after a certain number of steps for SDXL pipelines.
+- `IPAdapterScaleCutoffCallback`: Disables the IP Adapter after a certain number of steps.
+
+To set up a callback, you need to specify the number of steps after which the callback comes into effect. You can do so by using either one of these two arguments
+
+- `cutoff_step_ratio`: Float number with the ratio of the steps.
+- `cutoff_step_index`: Integer number with the exact number of the step.
+
+```python
+import torch
+from diffusers import StableDiffusionPipeline
+from diffusers.callbacks import SDCFGCutoffCallback
```

Comment:
Should we also show how to subclass from the `PipelineCallback` class and implement a implement a custom callback class?

Reply:
I can do it but then, what would be the difference with the regular callbacks? Also this is a little bit harder to do, would need more code and has more restrictions than just creating a callback function.

Normal callback:
```python
def callback_dynamic_cfg(pipe, step_index, timestep, callback_kwargs):
        if step_index == int(pipeline.num_timesteps * 0.4):
                prompt_embeds = callback_kwargs["prompt_embeds"]
                prompt_embeds = prompt_embeds.chunk(2)[-1]

                pipeline._guidance_scale = 0.0
                callback_kwargs["prompt_embeds"] = prompt_embeds
        return callback_kwargs
```

PipelineCallback subclass:
```python
class SDCFGCutoffCallback(PipelineCallback):
    tensor_inputs = ["prompt_embeds"]

    def callback_fn(self, pipeline, step_index, timestep, callback_kwargs) -> Dict[str, Any]:
        cutoff_step_ratio = self.config.cutoff_step_ratio
        cutoff_step_index = self.config.cutoff_step_index

        cutoff_step = (
            cutoff_step_index if cutoff_step_index is not None else int(pipeline.num_timesteps * cutoff_step_ratio)
        )

        if step_index == cutoff_step:
            prompt_embeds = callback_kwargs[self.tensor_inputs[0]]
            prompt_embeds = prompt_embeds[-1:]

            pipeline._guidance_scale = 0.0

            callback_kwargs[self.tensor_inputs[0]] = prompt_embeds
        return callback_kwargs
```

If you think it's still worth it, I can do it without a problem.

Reply:
Okay, thanks for explaining. I don't think there's a need then.